### ISSUE TITLE
character(len=*) for non dummy variables to be read in.

### ISSUE THREAD

**Issue by wsmorgan** [rosenbrockc/fortpy#58]
When trying to read in a string the compiler naturally doesn't know what the size of said string will be, so naturally the variable is initially given (len=*) so that the shape can be defined later. For non-dummy parameters, however, this fails:
```
ucase.f90:9.28:

  character(len=*) :: string
                            1
Error: Entity with assumed character length at (1) must be a dummy argument or a PARAMETER
gfortran: error: ucase.o: No such file or directory
```
As is explained here, http://stackoverflow.com/questions/25051108/fortran-character-len-only-apply-to-dummy-argument-or-parameter, the compiler can't determine shape of the variable since it isn't being passed in but gets read in after the variable has been set up. We may need to look at another option such as those recommended here, http://stackoverflow.com/questions/14765382/fortran-95-reading-a-character-string-of-unknown-length, in order to get around this.

**Comment by rosenbrockc** [comment]
This is a case where you need to override the default kind of the `character` variable using a `<global>` tag. When `fpy_read` gets values from the file, it needs to know in advance how large the character arrays are (i.e. strings) that the values will be read into. I considered just using a default value, but some input/output files have strings that are really long, so I decided against it.

The only enhancement to fortpy I can think of is to throw an error if a person doesn't explicitly declare the size of the character arrays that are being read in automatically. Or we could use `character(100)` as the default and then force override with `<global>` only when that is not sufficient. Thoughts?

**Comment by wsmorgan** [comment]
Having the error would be useful, using a global to override the value to
be a certain size should work to. I a default would be nice but it could
also create a whole bunch of new issues for the user.

Wiley Morgan
http://hagrid.byu.edu/Wiley#Bio

On Wed, Jun 17, 2015 at 1:44 PM, Conrad Rosenbrock <notifications@github.com

**Comment by rosenbrockc** [comment]
Not really. If the variable is re-used anywhere else (for example in pre-req routines) then its size would already be known to fortpy and this wouldn't be an issue at all. The problem is raised when the only routine that consumes the variable has also declared it as assumed-length. In that case, as long as it is long enough, they shouldn't run into any errors.

Perhaps the correct way to handle this is to issue a warning saying that a default size of 100 is being used since only assumed-length declarations are available in the methods that consume the variable.

**Comment by wsmorgan** [comment]
That should work, it'll handle most of the UNCLE code too.

Wiley Morgan
http://hagrid.byu.edu/Wiley#Bio

On Wed, Jun 17, 2015 at 3:14 PM, Conrad Rosenbrock <notifications@github.com

**Comment by rosenbrockc** [comment]
This should be fixed in Revision 1.6.1.
